src/storage_tests.cc

```diff
@@ -869,6 +872,113 @@ TEST_F(StorageTest, CCap_Inventory) {
   EXPECT_EQ(1, qr.GetVal<double>("Quantity", 4));
 }
 
+TEST_F(StorageTest, PackageExactly) {
+  std::string config =
+    "   <in_commods> <val>spent_fuel</val> </in_commods> "
+    "   <out_commods> <val>dry_spent</val> </out_commods> "
+    "   <throughput>10</throughput> "
+    "   <package>foo</package>";
+
+  int simdur = 3;
+
+  cyclus::MockSim sim(cyclus::AgentSpec (":cycamore:Storage"), config, simdur);
+  sim.context()->AddPackage("foo", 1, 2, "first");
+  cyclus::Package::Ptr p = sim.context()->GetPackage("foo");
+
+  sim.AddSource("spent_fuel").capacity(2).Finalize();
+  sim.AddSink("dry_spent").Finalize();
+
+  int id = sim.Run();
+
+  std::vector<cyclus::Cond> tr_conds;
+  tr_conds.push_back(cyclus::Cond("Commodity", "==", std::string("dry_spent")));
+  cyclus::QueryResult qr_trans = sim.db().Query("Transactions", &tr_conds);
+  EXPECT_EQ(2, qr_trans.rows.size());
+
+  EXPECT_EQ(1, qr_trans.GetVal<int>("Time", 0));
+  EXPECT_EQ(2, qr_trans.GetVal<int>("Time", 1));
+
+  std::vector<cyclus::Cond> res_conds;
+  res_conds.push_back(cyclus::Cond("PackageName", "==", p->name()));
+  cyclus::QueryResult qr_res = sim.db().Query("Resources", &res_conds);
+  EXPECT_EQ(qr_res.rows.size(), 2);
+  // Given the PRNG with default seed, the resource should have mass 9.41273
```

Comment:
It's not obvious why we have random behavior here, or how these tests confirm this pseudo-random behavior.

Reply:
Oops, we don't. Bad copy-paste mistake on my part